modules/tensor_mechanics/src/materials/ADTransverselyIsotropicPlasticityStressUpdate.C

```diff
@@ -0,0 +1,353 @@
+
+  omega_gamma /= 4.0 * omega;
+  sy_gamma = 2.0 * sy_alpha * (omega + delta_gamma * omega_gamma);
+}
+
+Real
+ADTransverselyIsotropicPlasticityStressUpdate::computeReferenceResidual(
+    const ADDenseVector & /*effective_trial_stress*/,
+    const ADDenseVector & /*stress_new*/,
+    const ADReal & /*residual*/,
+    const ADReal & /*scalar_effective_inelastic_strain*/)
+{
+  // Equation is already normalized.
+  return 1.0;
+}
+
+ADReal
+ADTransverselyIsotropicPlasticityStressUpdate::computeResidual(
+    const ADDenseVector & stress_dev,
+    const ADDenseVector & /*stress_sigma*/,
+    const ADReal & delta_gamma)
+{
+
+  // If in elastic regime, just return
+  if (_yield_condition <= 0.0)
+    return 0.0;
+
+  ADDenseMatrix eigenvectors_hill_transpose(6, 6);
+
+  _eigenvectors_hill.get_transpose(eigenvectors_hill_transpose);
+  eigenvectors_hill_transpose.vector_mult(_stress_np1, stress_dev);
+
+  ADReal omega = computeOmega(delta_gamma, _stress_np1);
+  _hardening_slope = computeHardeningDerivative();
+
+  // Hardening variable is \alpha isotropic hardening for now.
+  _hardening_variable[_qp] = computeHardeningValue(delta_gamma, omega);
+  ADReal s_y = _hardening_slope * _hardening_variable[_qp] + _yield_stress;
+
+  ADReal residual = 0.0;
+  residual = s_y / omega - 1.0;
+
+  return residual;
+}
+
+ADReal
+ADTransverselyIsotropicPlasticityStressUpdate::computeDerivative(
+    const ADDenseVector & /*stress_dev*/,
+    const ADDenseVector & /*stress_sigma*/,
+    const ADReal & delta_gamma)
+{
+  // If in elastic regime, return unit derivative
+  if (_yield_condition <= 0.0)
+    return 1.0;
+
+  ADReal omega = computeOmega(delta_gamma, _stress_np1);
+  _hardening_slope = computeHardeningDerivative();
+
+  ADReal sy = _hardening_slope * computeHardeningValue(delta_gamma, omega) + _yield_stress;
+  ADReal sy_alpha = _hardening_slope;
+
+  ADReal omega_gamma;
+  ADReal sy_gamma;
+
+  computeDeltaDerivatives(delta_gamma, _stress_np1, sy_alpha, omega, omega_gamma, sy_gamma);
+  ADReal residual_derivative = 1 / omega * (sy_gamma - 1 / omega * omega_gamma * sy);
+
+  return residual_derivative;
+}
+
+void
+ADTransverselyIsotropicPlasticityStressUpdate::computeHillTensorEigenDecomposition(
+    ADDenseMatrix & hill_tensor)
+{
+  unsigned int dimension = hill_tensor.n();
+
+  AnisotropyMatrix A;
+  for (unsigned int index_i = 0; index_i < dimension; index_i++)
+    for (unsigned int index_j = 0; index_j < dimension; index_j++)
+      A(index_i, index_j) = hill_tensor(index_i, index_j);
+
+  Eigen::SelfAdjointEigenSolver<AnisotropyMatrix> es(A);
+
+  auto lambda = es.eigenvalues();
+  auto v = es.eigenvectors();
+  for (unsigned int index_i = 0; index_i < dimension; index_i++)
+    _eigenvalues_hill(index_i) = lambda(index_i);
+
+  for (unsigned int index_i = 0; index_i < dimension; index_i++)
+    for (unsigned int index_j = 0; index_j < dimension; index_j++)
+      _eigenvectors_hill(index_i, index_j) = es.eigenvectors()(index_i, index_j);
```

Comment:
```suggestion
      _eigenvectors_hill(index_i, index_j) =v(index_i, index_j);
```

Reply:
I didn't commit your suggestion from here because I was afraid it was going to have a formatting issue